main.c

```diff
@@ -343,25 +326,35 @@ void setProfile() {
  * Execute current profile
  */
 void executeProfile() {
-  chSysLock();
   // Here we disable the foreground to ensure the animation will run
   is_foregroundColor_set = false;
 
   if (profiles[currentProfile].profileInit != NULL) {
     profiles[currentProfile].profileInit(ledColors);
   }
   profiles[currentProfile].callback(ledColors, ledIntensity);
-  chSysUnlock();
 }
 
 /*
  * Turn off all leds
  */
 void disableLeds() {
+  ledEnabled = false;
   palClearLine(LINE_LED_PWR);
-  if (GPTD_BFTM1.state == GPT_CONTINUOUS) {
-    gptStopTimer(&GPTD_BFTM1);
+
+  // gptChangeInterval is unreliable, restarting because of that
+  gptStopTimer(&GPTD_BFTM0);
+  gptStartContinuous(&GPTD_BFTM0, SERIAL_CHECK_FREQUENCY);
```

Comment:
When LEDs are disabled, there is no reason to call the callback as frequently as before, so slowing it down.
When LEDs are reenabled, previous frequency 1 tick is reenabled.